src/lib/realmdb/RealmDB.js

```diff
@@ -300,6 +348,59 @@ class RealmDB implements DB {
     log.debug('getFeedPage result:', numResults, offset, res.length, res)
     return res
   }
+
+  //TODO:  make sure profile contains walletaddress or enforce it in schema in realmdb
+  setProfile(profile) {
```

Comment:
1. please add correct types for the all method you've missed them
2. all mongo functions (except watch) are returning promise. so the methods **should** be defined as async